import "ydb/public/api/protos/ydb_import.proto";
import "ydb/public/api/protos/ydb_issue_message.proto";
import "ydb/public/api/protos/ydb_operation.proto";
import "ydb/public/api/protos/ydb_status_codes.proto";

import "google/protobuf/timestamp.proto";

package NKikimrImport;
option java_package = "ru.yandex.kikimr.proto";

message TImport {
    optional uint64 Id = 1;
    optional Ydb.StatusIds.StatusCode Status = 2;
    repeated Ydb.Issue.IssueMessage Issues = 3;
    optional Ydb.Import.ImportProgress.Progress Progress = 4;
    optional google.protobuf.Timestamp StartTime = 7;
    optional google.protobuf.Timestamp EndTime = 8;
    repeated Ydb.Import.ImportItemProgress ItemsProgress = 5;
    oneof Settings {
        Ydb.Import.ImportFromS3Settings ImportFromS3Settings = 6;
    }
    optional string UserSID = 9;
}

message TCreateImportRequest {
    optional Ydb.Operations.OperationParams OperationParams = 1;
    oneof Settings {
        Ydb.Import.ImportFromS3Settings ImportFromS3Settings = 2;
    }
}

message TEvCreateImportRequest {
    optional uint64 TxId = 1;
    optional string DatabaseName = 2;
    optional string UserSID = 3;
    optional TCreateImportRequest Request = 4;
    optional string PeerName = 5;
}

message TCreateImportResponse {
    optional TImport Entry = 1;
}

message TEvCreateImportResponse {
    optional uint64 TxId = 1;
    optional TCreateImportResponse Response = 2;
}

message TGetImportRequest {
    optional uint64 Id = 1;
}

message TEvGetImportRequest {
    optional string DatabaseName = 2;
    optional TGetImportRequest Request = 1;
}

message TGetImportResponse {
    optional TImport Entry = 1;
}

message TEvGetImportResponse {
    optional TGetImportResponse Response = 1;
}

message TCancelImportRequest {
    optional uint64 Id = 1;
}

message TEvCancelImportRequest {
    optional uint64 TxId = 1;
    optional TCancelImportRequest Request = 2;
    optional string DatabaseName = 3;
    optional string UserSID = 4;
    optional string PeerName = 5;
}

message TCancelImportResponse {
    optional Ydb.StatusIds.StatusCode Status = 1;
    repeated Ydb.Issue.IssueMessage Issues = 2;
}

message TEvCancelImportResponse {
    optional uint64 TxId = 1;
    optional TCancelImportResponse Response = 2;
}

message TForgetImportRequest {
    optional uint64 Id = 1;
}

message TEvForgetImportRequest {
    optional uint64 TxId = 1;
    optional TForgetImportRequest Request = 2;
    optional string DatabaseName = 3;
    optional string UserSID = 4;
    optional string PeerName = 5;
}

message TForgetImportResponse {
    optional Ydb.StatusIds.StatusCode Status = 1;
    repeated Ydb.Issue.IssueMessage Issues = 2;
}

message TEvForgetImportResponse {
    optional uint64 TxId = 1;
    optional TForgetImportResponse Response = 2;
}

message TListImportsRequest {
    optional uint64 PageSize = 1;
    optional string PageToken = 2;
    optional string Kind = 3;
}

message TEvListImportsRequest {
    optional string DatabaseName = 2;
    optional TListImportsRequest Request = 1;
}

message TListImportsResponse {
    optional Ydb.StatusIds.StatusCode Status = 1;
    repeated Ydb.Issue.IssueMessage Issues = 2;
    repeated TImport Entries = 3;
    optional string NextPageToken = 4;
}

message TEvListImportsResponse {
    optional TListImportsResponse Response = 1;
}
